app.rb

```diff
@@ -0,0 +1,143 @@
+      person.rentals.each { |rental| puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}" }
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [input the number]:'
+    type = gets.chomp.to_i
+    case type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    end
+  end
+
+  def create_student
+    print 'Age:'
+    age = gets.chomp.to_i
+    print 'Name:'
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]:'
+    permission = gets.chomp.downcase
+    parent_permission = true
+    case permission
+    when 'n'
+      parent_permission = false
+    when 'y'
+      parent_permission = true
+    else
+      puts 'Out of range!'
+    end
+    @people << Student.new(age, nil, name, parent_permission: parent_permission)
+    puts 'you have successfully registered a student!'
+  end
+
+  def create_teacher
+    print 'Age:'
+    age = gets.chomp.to_i
+    print 'Name:'
+    name = gets.chomp
+    print 'Specialization:'
+    specialization = gets.chomp
+    @people << Teacher.new(age, specialization, name)
+    puts 'you have successfully registered a teacher!'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    puts 'Book created successfully'
+    @books << Book.new(title, author)
+  end
+
+  def create_rental
+    return puts 'No book to rent!' if @books.empty?
+    return puts 'No person found To rent a book!' if @people.empty?
+
+    book_number = select_book
+    return if book_number.nil?
+
+    person_number = select_person
+    return if person_number.nil?
+
+    date = get_user_input('Enter date [YYYY-MM-DD]')
+    return if date.nil?
+
+    rental = Rental.new(date, @books[book_number], @people[person_number])
+    save_rental_data(@books[book_number], @people[person_number], rental)
+  end
+
+  private
+
+  def select_book
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+    book_number = gets.chomp.to_i
+    return nil if book_number >= @books.length
+
+    book_number
+  end
+
+  def select_person
+    puts 'Select a person from the following list by number'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_number = gets.chomp.to_i
+    return nil if person_number >= @people.length
+
+    person_number
+  end
+
+  def get_user_input(message)
+    puts message
+    gets.chomp
+  end
+
+  def save_rental_data(book, person, rental)
+    person.rent_book(rental)
+    book.add_rentals(rental)
+    puts 'Rental created successfuly'
+  end
+
+  # app = App.new()
+  # app.create_person
+  # app.list_people
+  # app.create_book
+  # app.list_books
+  # app.create_rental
+  # app.list_rentals_by_person_id
```

Comment:
- [ ] To improve code readability and ensure that only active code is present, please remove the commented code from your codebase. This practice helps reduce confusion and makes it easier for others to understand and maintain the code in the future. ✨📌